x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -96,13 +98,28 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 	}
 
 	// Get listMetricsTotal and namespaces from configuration
-	listMetricsTotal, namespacesTotal := readCloudwatchConfig(m.CloudwatchMetrics.CloudwatchConfigs)
+	listMetricsWithRegion, listMetricsWithoutRegion, namespacesTotal := readCloudwatchConfig(m.CloudwatchConfigs)
+	if listMetricsWithoutRegion != nil {
+		m.Logger().Info("cloudwatch_metrics config is missing region name. " +
+			"To avoid extra costs, please make sure region is set in config.yml")
+	}
 
-	// Use listMetricsTotal from config
-	svcCloudwatch := cloudwatch.New(*m.MetricSet.AwsConfig)
-	err = createEvents(svcCloudwatch, listMetricsTotal, m.PeriodInSec, startTime, endTime, report)
-	if err != nil {
-		return errors.Wrap(err, "createEvents failed")
+	for regionName, listMetricsPerRegion := range listMetricsWithRegion {
+		m.MetricSet.AwsConfig.Region = regionName
```

Comment:
This is modifying the config in the metricset, this can be error-prone, we can leave the config unexpectedly modified, or we can have problems if we do something concurrent at some moment.
I think we could copy the config before modifying it, something like:
```suggestion
		awsConfig := *m.MetricSet.AwsConfig
		awsConfig.Region = regionName
		svcCloudwatch := cloudwatch.New(awsConfig)
```
The same in other places here where we also set the region.

Reply:
Yeah good point, I need to make this change in other aws metricsets. I will push a separate PR for this! Thank you!